src/lib/utils.py

```diff
@@ -113,24 +132,24 @@ def extract_appimage_version(basename, data):
             fp.write(data)
 
         os.chmod(appimage_path, 0o755)
-        args = ["bwrap"]
-        for path in ("/usr", "/lib", "/lib64", "/bin", "/proc"):
-            args.extend(["--ro-bind", path, path])
-        args.extend(
-            [
-                "--bind",
-                tmpdir,
-                tmpdir,
-                "--die-with-parent",
-                "--new-session",
-                "--unshare-all",
-                appimage_path,
-                "--appimage-extract",
-            ]
-        )
+        args = []
+        if _check_bwrap():
+            args.extend(
+                [
+                    "--bind",
+                    tmpdir,
+                    tmpdir,
+                    "--die-with-parent",
+                    "--new-session",
+                    "--unshare-all"
+                ]
+            )
+
+        args.extend([appimage_path,"--appimage-extract"])
+
         log.debug("$ %s", " ".join(args))
+        p = bwrap_run(args, cwd=tmpdir)
```

Comment:
Now you'll try to run `bwrap` even on the path where you've determined it's not present :(

Reply:
I am a bit concerned about executing arbitrary code, with network access, in an environment that has access to (at least) an API key that can push to this repo. I suppose you could argue that it's the same arbitrary code that the end-user will run on their actual computer anyway…

Reply:
> Now you'll try to run bwrap even on the path where you've determined it's not present :(

Reply:
> I am a bit concerned about executing arbitrary code, with network access, in an environment that has access to (at least) an API key that can push to this repo

It's valid concern, but on the other hand, it's not something I can really change. I really want to avoid extending Flathub's buildbot to accommodate anything beyond building applications and basic post-build checks.

I guess I could switch to using https://github.com/refi64/unappimage for extraction.

Reply:
Yeah, it's probably quite low risk in practice. Perhaps open an issue for trying to switch to unappimage in futuer?